File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    Map = [[' '] * n for _ in range(n)]
    answer = []
    
    for i in range(n):
        tmp = ''
        Bin1 = list(bin(arr1[i])[2:])
        Bin2 = list(bin(arr2[i])[2:])
        
        for j in range(len(Bin1)-1,-1,-1):
            if Bin1[j] == '1':Map[i][len(Bin1)-1-j] = '#'
        
        for j in range(len(Bin2)-1,-1,-1):
            if Bin2[j] == '1':Map[i][len(Bin2)-1-j] = '#'
        
        for j in range(n-1,-1,-1):
            tmp += Map[i][j]
        answer.append(tmp)
    return answer

# n = 5
# arr1 = [9,20,28,18,11]
# arr2 = [30,1,21,17,28]

# print(solution(n,arr1,arr2))

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const Map = arr1.map((num1,idx) => {
        bin1 = num1.toString(2).padStart(n, "0").split("")
        bin2 = arr2[idx].toString(2).padStart(n, "0").split("")
        
        password = ""
        for (let i = 0; i < n; i++){
            if(bin1[i] === '0' && bin2[i] === '0'){password += " "}
            else {password += "#"}
        }
        return password
    })
    return Map
}

// function solution(n, arr1, arr2) {
//     let G = Array.from(Array(n), () => Array())
    
//     for(let i = 0; i < n; i++){
//         bin1 = arr1[i].toString(2)
//         bin2 = arr2[i].toString(2)
//         if(bin1.length < n) {bin1 = ("0".repeat(n - (bin1.length))) + bin1}
//         if(bin2.length < n) {bin2 = ("0".repeat(n - (bin2.length))) + bin2}
        
//         for(let j = 0; j < n; j++){
//             if(bin1[j] === "0" && bin2[j] === "0") {
//                 G[i] += " "
//             }
//             else {
//                 G[i] += "#" 
//             }
//         }
//     }
//     console.log(G)
//     return G;
// }